### PR TITLE
Fix equalize unit test

### PR DESCRIPTION
**Pull request by stiepan**
## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
Fixes a bug in equalize unit tests. One of the cases assumed the allocated memory is zero-ed out, while it is not.

### Tests:

- [ ] Existing tests apply
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [x] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by stiepan** [comment]
!build

File: dali/kernels/imgproc/color_manipulation/equalize/lut_test.cc [hunk 1]
```diff
@@ -163,8 +163,11 @@ TEST_F(EqualizeLutGpuTest, SinglePoint) {
   auto in_view = in_.cpu();
   for (int sample_idx = 0; sample_idx < batch_shape.num_samples(); sample_idx++) {
     for (int channel_idx = 0; channel_idx < batch_shape[sample_idx][0]; channel_idx++) {
-      in_view[sample_idx].data[channel_idx * range_size + sample_idx * 71 + channel_idx] =
-          123 * channel_idx + sample_idx;
+      ASSERT_EQ(batch_shape[sample_idx][1], range_size);
+      for (int i = 0; i < range_size; i++) {
+        in_view[sample_idx].data[channel_idx * range_size + i] =
+            (i == 71 + channel_idx) ? ((sample_idx << 11) + channel_idx) : 0;
```
Comment (mzient) [inline dali/kernels/imgproc/color_manipulation/equalize/lut_test.cc:169]: This code looks almost as if intentionally obfuscated.
```suggestion
        int index_of_nonzero_value = 71 + channel_idx;
        int nonzero_value = (sample_idx << 11) + channel_idx;
        in_view[sample_idx].data[channel_idx * range_size + i] =
            i == index_of_nonzero_value ? nonzero_value : 0;
```

Comment (stiepan) [reply dali/kernels/imgproc/color_manipulation/equalize/lut_test.cc]: fixed

**Comment by stiepan** [comment]
!build

File: dali/kernels/imgproc/color_manipulation/equalize/lut_test.cc [hunk 2]
```diff
@@ -163,8 +163,13 @@ TEST_F(EqualizeLutGpuTest, SinglePoint) {
   auto in_view = in_.cpu();
   for (int sample_idx = 0; sample_idx < batch_shape.num_samples(); sample_idx++) {
     for (int channel_idx = 0; channel_idx < batch_shape[sample_idx][0]; channel_idx++) {
-      in_view[sample_idx].data[channel_idx * range_size + sample_idx * 71 + channel_idx] =
-          123 * channel_idx + sample_idx;
+      ASSERT_EQ(batch_shape[sample_idx][1], range_size);
+      for (int i = 0; i < range_size; i++) {
+        int index_of_nonzero_value = 71 + channel_idx;
```
Comment (mzient) [inline dali/kernels/imgproc/color_manipulation/equalize/lut_test.cc:168]:
```suggestion
        int index_of_nonzero_value = sample_idx * 71 + channel_idx;
```

**Review by mzient — APPROVED** [review]

File: dali/kernels/imgproc/color_manipulation/equalize/lut_test.cc [hunk 3]
```diff
@@ -163,8 +163,14 @@ TEST_F(EqualizeLutGpuTest, SinglePoint) {
   auto in_view = in_.cpu();
   for (int sample_idx = 0; sample_idx < batch_shape.num_samples(); sample_idx++) {
     for (int channel_idx = 0; channel_idx < batch_shape[sample_idx][0]; channel_idx++) {
-      in_view[sample_idx].data[channel_idx * range_size + sample_idx * 71 + channel_idx] =
-          123 * channel_idx + sample_idx;
+      ASSERT_EQ(batch_shape[sample_idx][1], range_size);
+      int nonzero_idx = 71 + channel_idx;
```
[the remark this answers is not in the document]
Comment (stiepan) [reply dali/kernels/imgproc/color_manipulation/equalize/lut_test.cc]: After all, why not. I added it back.

**Comment by stiepan** [comment]
!build

**Review by klecki — APPROVED** [review]

**Pull request merged**